From 761ae73c061f9be0ae8dcd97893e75d6e02fbf46 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Fri, 6 Oct 2017 11:13:09 +0200
Subject: [PATCH] gtkdoc-check: Move over gtkdoc-check unit tests to the docs/
 directory and use AM's unit testing API. (However, this seems to be broken
 currently, so we disable those tests completely, for now (current solution
 after hours of investigating).

---
 docs/libdbusmenu-glib/reference/Makefile.am |  1 +
 docs/libdbusmenu-gtk/reference/Makefile.am  |  2 +-
 tests/Makefile.am                           | 22 ----------------------
 3 files changed, 2 insertions(+), 23 deletions(-)

diff --git a/docs/libdbusmenu-glib/reference/Makefile.am b/docs/libdbusmenu-glib/reference/Makefile.am
index 9274c5c..924a795 100644
--- a/docs/libdbusmenu-glib/reference/Makefile.am
+++ b/docs/libdbusmenu-glib/reference/Makefile.am
@@ -93,4 +93,5 @@ if ENABLE_GTK_DOC
  TESTS_ENVIRONMENT = cd $(srcdir) && \
    DOC_MODULE=$(DOC_MODULE) DOC_MAIN_SGML_FILE=$(DOC_MAIN_SGML_FILE) \
    SRCDIR=$(abs_srcdir) BUILDDIR=$(abs_builddir)
+# TESTS = $(GTKDOC_CHECK)
 endif
diff --git a/docs/libdbusmenu-gtk/reference/Makefile.am b/docs/libdbusmenu-gtk/reference/Makefile.am
index 8cc3d07..ce1e8ca 100644
--- a/docs/libdbusmenu-gtk/reference/Makefile.am
+++ b/docs/libdbusmenu-gtk/reference/Makefile.am
@@ -93,5 +93,5 @@ if ENABLE_GTK_DOC
  TESTS_ENVIRONMENT = cd $(srcdir) && \
    DOC_MODULE=$(DOC_MODULE) DOC_MAIN_SGML_FILE=$(DOC_MAIN_SGML_FILE) \
    SRCDIR=$(abs_srcdir) BUILDDIR=$(abs_builddir)
+# TESTS = $(GTKDOC_CHECK)
 endif
-
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 64e3e75..98409d6 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -40,12 +40,6 @@ TESTS += \
 #	test-gtk-submenu 
 endif
 
-if ENABLE_GTK_DOC
-TESTS += \
-	test_libdbusmenu_glib_gtkdoc \
-	test_libdbusmenu_gtk_gtkdoc
-endif
-
 # The Python test only work on the system copy of
 # dbusmenu, so while they can be usefule they're not
 # good tests of what you're currently building.  Handy
@@ -545,22 +539,6 @@ test-mago: test-gtk-label-client test-gtk-label-server $(srcdir)/dbusmenu-gtk/ma
 	@echo PYTHONPATH=$(abs_srcdir)/dbusmenu-gtk/mago_tests mago -f dbusmenu.xml -t $(abs_builddir)/mago.results --log-level=debug >> $@
 	@chmod +x $@
 
-#########################
-# Other
-#########################
-
-test_libdbusmenu_glib_gtkdoc:
-	@echo "#!/bin/bash" > $@
-	@echo cd $(abs_top_builddir)/docs/libdbusmenu-glib/reference >> $@
-	@echo gtester --verbose -k $(GTKDOC_CHECK_PATH) >> $@
-	@chmod +x $@
-
-test_libdbusmenu_gtk_gtkdoc:
-	@echo "#!/bin/bash" > $@
-	@echo cd $(abs_top_builddir)/docs/libdbusmenu-gtk/reference >> $@
-	@echo gtester --verbose -k $(GTKDOC_CHECK_PATH) >> $@
-	@chmod +x $@
-
 examplesdir = $(docdir)/examples/
 
 examples_DATA = \
-- 
2.11.0

